lib/galaxy/util/properties.py

```diff
@@ -87,13 +87,20 @@ def load_app_properties(
     # update from env
     override_prefix = "%sOVERRIDE_" % config_prefix
     for key in os.environ:
+        config_key = None
         if key.startswith(override_prefix):
             config_key = key[len(override_prefix):].lower()
             properties[config_key] = os.environ[key]
         elif key.startswith(config_prefix):
             config_key = key[len(config_prefix):].lower()
             if config_key not in properties:
                 properties[config_key] = os.environ[key]
+        try:
+            # Attempt to parse value as yaml to allow passing complex options via env
+            if config_key:
+                properties[config_key] = yaml.safe_load(properties[config_key])
+        except yaml.YAMLError:
+            pass
```

Comment:
I think this should be applied only to `GALAXY_CONFIG_OVERRIDE_`, not `GALAXY_CONFIG_`.

Reply:
How so?
If I want to include complex configuration via env only if it is not already present in the galaxy.yml, this would need to work for both.